### ISSUE TITLE
Allow disabling positional audio for team members

### ISSUE THREAD

**Issue by fenhl** [mumble-voip/mumble#4387]
**Context**
This could provide the basis for a game variant in team-based competitions: depending on the game organizer's preference, things could be set up so that teammates can always hear each other, while still having the ability to locate opponents who are talking near them.

**Describe the feature you have in mind**
Positional audio plugins should have the ability to group users such that users in the same group can always hear each other regardless of distance, while using positional audio as normal for users in other groups.

**Describe alternatives you've considered**
An additional Mumble instance without positional audio or a different voice chat service could be used to ensure that teammates can always hear each other, but this would result in doubled audio when teammates are near each other.

**Additional context**
Essentially, a game set up this way would emulate player characters being in conference calls inside the game world, where you're only talking to your team but others nearby can still hear you talk.

I am hoping to get this feature added to [the Minecraft mod](https://github.com/magneticflux-/fabric-mumblelink-mod) once it is supported by Mumble itself.

**Comment by Krzmbrzl** [comment]
I'm not sure this is a feature that is needed/wanted by many people...

Alternatively with 1.4.0 you could set up 2 channels for the teams and let everyone of a given team listen to their respective channel. Then you can set up a shortcut to shout to that channel and to ignore positional audio while doing so.
This way you can use positional audio as normal but you can use the shortcut as a way to shout to your entire team regardless of their ingame location.

Tbh this actually sounds much more enjoyable to me as you don't have to constantly hear everyone of your team which - depending on the team's size - could be really messy.

**Comment by Kissaki** [comment]
The game knows about the teams so it will have to provide that information.

Our current implementation provides context and identity where context is the context people hear each other in. This us currently limited to the game/all PA participants.

For a two-level differentiation this is not sufficient.

**Comment by Krzmbrzl** [comment]
And I suppose that requiring plugins to also reverse-engineer a way to figure out the team a player is in, is way too much (given that I expect this to be a lot harder than figuring out the memory addresses of the player's position).

**Comment by fenhl** [comment]
It's not hard if you use a game mod (as is the case for Minecraft), or you're a game developer integrating support into the game itself.

**Comment by Krzmbrzl** [comment]
Yeah but we can't assume all our plugins to be integrated into the game via a Mod or in the game itself. We have to make this work for the general case and not just for this specific one.

One thing we could do about that is making this kind of information optional and only use it if available. This still leaves the question whether it is that valuable in the first place or if a workaround similar to the one I described above is sufficient for the presumably few cases in which this is needed :thinking:

**Comment by Kissaki** [comment]
We would not have to support it for every PA game at once. It could be an additional feature for specific games.

**Comment by Krzmbrzl** [comment]
Yeah that's kinda what I meant with making the info optional. But the question whether it is worth the effort still remains... Especially given that I think the wanted behavior can be achieved using the new ChannelListener featurein 1.4.0 :shrug:

**Comment by fenhl** [comment]
What kind of info is needed?

**Comment by Krzmbrzl** [comment]
Whether the workaround described above is sufficient for your case.
‘cause i stilk don’t think that implementing thid is worth the effort.

**Comment by fenhl** [comment]
Assuming you mean https://github.com/mumble-voip/mumble/issues/4387#issuecomment-667478349, no, that is not sufficient, since it requires actively pressing a key to talk to your team, which is infeasible during a game.

**Comment by Krzmbrzl** [comment]
Yes that's what I meant. Hm okay, I guess since I'm more into slow-paced games having to press a button doesn't bother me but I can see how it could get problematic in a fast-paced game...

**Comment by Kissaki** [comment]
We talked about this in our team meeting last weekend.

We can see the use case and deem this feature request valid and would like to see this implemented. Implementing this would require extending the PA context logic considerably. The potential userbase and consequently impact is pretty small and we have many other things to work on so for the foreseeable future none of us will be able to work on this. We are not opposed to community contributions though. From what others said it’s not an easy task though considering existing logic and having to adjust and extend it.
